### PR TITLE
WFLY-18508 spring-resteasy Quickstart Common Enhancements CY2023Q3

### PR DESCRIPTION
**Pull request by liweinan**
https://issues.redhat.com/browse/WFLY-18508

### CODE REVIEW

**Comment by liweinan** [comment]
@emmartins @jamezp I start to work on this according to the changes in: https://github.com/wildfly/quickstart/pull/711

It's still in draft and I'll complete the missing parts. Currently the integration-test and existing tests can pass.

**Comment by liweinan** [comment]
The JDK11 can not be used because this example needs at least JDK17.

File: spring-resteasy/pom.xml [hunk 1]
```diff
@@ -50,7 +50,12 @@
 
         <!-- The versions for BOMs, Dependencies and Plugins -->
         <version.spring.framework>6.0.4</version.spring.framework>
-        <version.server.bom>30.0.0.Final</version.server.bom>
+        <!-- the version for the Server -->
+        <version.server>30.0.0.Final</version.server>
+        <!-- the versions for BOMs, Packs and Plugins -->
+        <version.bom.ee>${version.server}</version.bom.ee>
+        <version.pack.cloud>5.0.0.Beta1</version.pack.cloud>
```
Comment (emmartins) [inline spring-resteasy/pom.xml:57]: 5.0.0.Final

File: spring-resteasy/pom.xml [hunk 2]
```diff
@@ -50,7 +50,12 @@
 
         <!-- The versions for BOMs, Dependencies and Plugins -->
         <version.spring.framework>6.0.4</version.spring.framework>
```
Comment (emmartins) [inline spring-resteasy/pom.xml:52]: please move this to the bottom and add the separator comment: 
```
<!-- The versions for other Dependencies and Plugins -->
<version.spring.framework>6.0.4</version.spring.framework>

```

File: spring-resteasy/pom.xml [hunk 3]
```diff
@@ -204,6 +209,11 @@
                         <groupId>org.apache.maven.plugins</groupId>
```
Comment (emmartins) [inline spring-resteasy/pom.xml:209]: I had a look at ResteasySpringIT.java and it's really just a set of http client tests so please get ridden of Arquillian (dependencies, arq-* profiles, arquillian.xml) and convert it to junit, working similar to BasicRuntimeTest, so we can add it to integration-testing.

**Review by emmartins — CHANGES_REQUESTED** [review]

**Comment by liweinan** [comment]
After removing the `arq` profiles the tests can be executed this way:

```bash
➤ mvn wildfly:run -Pprovisioned-server
```


```bash
➤ mvn verify -Pintegration-testing -Dserver.host=http://localhost:8080/                                                                                                                                                                                                                     00:04:11
```

<img width="2024" alt="image 2" src="https://github.com/wildfly/quickstart/assets/201907/3d0c3b57-3f69-4eee-8ca8-d4d0f41860ba">

<img width="1544" alt="image" src="https://github.com/wildfly/quickstart/assets/201907/a318f369-015f-46e2-9a7b-49c32810b6d8">


I will cleanup this PR and squash the commits after everything is fixed.

**Comment by emmartins** [comment]
> After removing the `arq` profiles the tests can be executed this way:
> 
> ```shell
> ➤ mvn wildfly:run -Pprovisioned-server
> ```
> 
> ```shell
> ➤ mvn verify -Pintegration-testing -Dserver.host=http://localhost:8080/                                                                                                                                                                                                                     00:04:11
> ```
> 
> <img alt="image 2" width="2000" src="https://user-images.githubusercontent.com/201907/277401466-3d0c3b57-3f69-4eee-8ca8-d4d0f41860ba.png"> <img alt="image" width="1544" src="https://user-images.githubusercontent.com/201907/277401515-a318f369-015f-46e2-9a7b-49c32810b6d8.png">
> I will cleanup this PR and squash the commits after everything is fixed.

ok seems I missed something, you didn't update the README.adoc as specified by:
* https://docs.google.com/document/d/1GFnZrnlwRkwC0HfziRws2UUWR99KdaQayA7Ei12SGN4/edit#heading=h.bksczecrc5z2
* https://docs.google.com/document/d/1GFnZrnlwRkwC0HfziRws2UUWR99KdaQayA7Ei12SGN4/edit#heading=h.fz37de8en4v  

The content include above provides you the testing instructions, which use wildfly:start -DjbossHome=... (as followup of mvn clean package  -Pprovisioned-server), which are the instructions being tested by the CI. I will consider a post-enhancements move to wildfly:run single command instead, but for now let's have all QS using same ok?

File: spring-resteasy/pom.xml [hunk 4]
```diff
@@ -217,22 +325,17 @@
...
-                </dependency>
-            </dependencies>
+            <id>integration-testing</id>
             <build>
                 <plugins>
                     <plugin>
                         <groupId>org.apache.maven.plugins</groupId>
                         <artifactId>maven-failsafe-plugin</artifactId>
-                        <version>${version.failsafe.plugin}</version>
+<!--                        <configuration>-->
```
Comment (emmartins) [inline spring-resteasy/pom.xml:334]: FYI what everyone is using elsewhere is to include the wildcard '**/*IT'

File: spring-resteasy/pom.xml [hunk 5]
```diff
@@ -186,29 +191,132 @@
     </dependencies>
```
Comment (emmartins) [inline spring-resteasy/pom.xml:192]: We need also plugin management to be added, please see https://docs.google.com/document/d/1GFnZrnlwRkwC0HfziRws2UUWR99KdaQayA7Ei12SGN4/edit#heading=h.rcftxwtudvl5

**Comment by jamezp** [comment]
Spring 6 only works on Java 17+. I'm not sure how we can resolve this because of the workflow call. Creating the workflow matrix in JSON is an option, but likely not a great one. However, the other option is to not call the workflow.

**Comment by liweinan** [comment]
> Spring 6 only works on Java 17+. I'm not sure how we can resolve this because of the workflow call. Creating the workflow matrix in JSON is an option, but likely not a great one. However, the other option is to not call the workflow.

Maybe we can split the CI file for Java 11 or 17? @emmartins wdyt?

File: spring-resteasy/pom.xml (hunk 5 above)
Comment (liweinan) [reply spring-resteasy/pom.xml:193]: @emmartins I have added the plugin management section, not sure I add it correctly or not. In addition I found the `batch-proessing`  example doesn't have this section either:

- https://github.com/wildfly/quickstart/blob/main/batch-processing/pom.xml

Shall I add the section their too?

**Comment by liweinan** [comment]
@emmartins Thanks for the advice! I'll update the README accordingly.

> > After removing the `arq` profiles the tests can be executed this way:
> > ```shell
> > ➤ mvn wildfly:run -Pprovisioned-server
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```shell
> > ➤ mvn verify -Pintegration-testing -Dserver.host=http://localhost:8080/                                                                                                                                                                                                                     00:04:11
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > <img alt="image 2" width="2000" src="https://user-images.githubusercontent.com/201907/277401466-3d0c3b57-3f69-4eee-8ca8-d4d0f41860ba.png"> <img alt="image" width="1544" src="https://user-images.githubusercontent.com/201907/277401515-a318f369-015f-46e2-9a7b-49c32810b6d8.png">
> > I will cleanup this PR and squash the commits after everything is fixed.
> 
> ok seems I missed something, you didn't update the README.adoc as specified by:
> 
> * https://docs.google.com/document/d/1GFnZrnlwRkwC0HfziRws2UUWR99KdaQayA7Ei12SGN4/edit#heading=h.bksczecrc5z2
> * https://docs.google.com/document/d/1GFnZrnlwRkwC0HfziRws2UUWR99KdaQayA7Ei12SGN4/edit#heading=h.fz37de8en4v
> 
> The content included above provides you the testing instructions, which use wildfly:start -DjbossHome=... (as followup of mvn package -Pprovisioned-server), which are the instructions being tested by the CI. I will consider a 'post-enhancements' move to wildfly:run single command instead, but for now let's have all QS using same ok?

File: spring-resteasy/pom.xml (hunk 5 above)
Comment (emmartins) [reply spring-resteasy/pom.xml:193]: Since this is not a MicroProfile quickstart you don't need the wildfly-jar plugin

Comment (emmartins) [reply spring-resteasy/pom.xml:193]: Wrt batch-processing it was fixed yesterday :-)

**Comment by emmartins** [comment]
@liweinan @jamezp no worries wrt the JDK, I will later update the quickstart_ci.yml to allow quickstarts to customize the matrix JDKs, I will do it elsewhere to avoid multiple runs of all QS workflows. For now please simply ignore the JDK 11 test fails.

**Comment by liweinan** [comment]
> @liweinan @jamezp no worries wrt the JDK, I will later update the quickstart_ci.yml to allow quickstarts to customize the matrix JDKs, I will do it elsewhere to avoid multiple runs of all QS workflows. For now please simply ignore the JDK 11 test fails.

Thanks!

**Comment by liweinan** [comment]
The tests passed locally with the following command: 


```bash
$ mvn clean package -Pprovisioned-server
```

```bash
$ mvn wildfly:start -DjbossHome=target/server
```

```bash
➤ mvn verify -Pintegration-testing
```

image

**Comment by liweinan** [comment]
@emmartins I have updated the doc and the tests passed in JDK17, so I mark this PR as ready for preview. Please let me know if there is anything else need to be changed :D

**Review by emmartins — CHANGES_REQUESTED** [review]
Looks good to me, only one more change, please squash the commits into one.

**Comment by liweinan** [comment]
@emmartins Done!

**Comment by emmartins** [comment]
thank you @liweinan

**Comment by emmartins** [comment]
FYI I created https://issues.redhat.com/browse/WFLY-18689 to deliver the matrix.jdk customization this QS needs. As I said before I will leave this to post-enhancements.

**Comment by emmartins** [comment]
@liweinan FYI [WFLY-18689](https://issues.redhat.com/browse/WFLY-18689) was now fixed, which included the changes in this quickstart yml to only test it on JDK 17

**Comment by liweinan** [comment]
> @liweinan FYI [WFLY-18689](https://issues.redhat.com/browse/WFLY-18689) was now fixed, which included the changes in this quickstart yml to only test it on JDK 17

Cool!

**Comment by liweinan** [comment]
Relative PR: https://github.com/wildfly/quickstart/pull/767

**Pull request merged**